apps/05Mastermind.js

```diff
@@ -29,19 +29,68 @@ function getRandomInt(min, max) {
   return Math.floor(Math.random() * (max - min)) + min;
 }
 
-function generateHint() {
+function generateHint(solution, guess) {
   // your code here
+
+  var solutionArray = solution.split('');
+  var guessArray = guess.split('');
+  var correctLetters = 0;
+  var correctLetterLocations;
+
+  for (var i = 0; i < solutionArray.length; i++) {
+    if (solutionArray.indexOf() === guessArray.indexOf()) {
+      solutionArray = null;
+      console.log("solution array", solutionArray[i]);
+
+    }
+  }
+
+  for (var i = 0; i < guessArray.length; i++) {
+    console.log("guess array", guessArray[i]);
+  }
 }
 
 function mastermind(guess) {
   // your code here
+  var counter = 0;
+
+  console.log(guess);
+  console.log(solution);
+  if (guess !== solution) {
+    board.push(guess);
+    console.log('board', board);
+    generateHint(solution, guess);
```

Comment:
You're off to a good start here.

Because hints are what fill up the board I would recommend calling the generateHint function first before pushing your hint to the board.